public_html/app/controllers/Posts.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+    class Posts extends Controller {
+
+        public function __construct() {
+            if (!isLoggedIn()) {
+                redirect('users/login');
+            }
+
+            $this->postModel = $this->model('Post');
+        }
+
+
+        // INDEX
+        public function index() {
+            // Get posts
+            $posts = $this->postModel->getPosts();
+
+            $data = [
+                'posts' => $posts
+            ];
+
+            $this->view('posts/index', $data);
+        }
+
+
+        // ADD
+        public function add() {
+
+            if($_SERVER['REQUEST_METHOD'] == 'POST') {
+                // Sanitize POST array
+                $_POST = filter_input_array(INPUT_POST, FILTER_SANITIZE_STRING);
+
+                $data = [
+                    'title' => trim($_POST['title']),
+                    'body' => trim($_POST['body']),
+                    'user_id' => $_SESSION['user_id'],
+                    'title_error' => '',
```

Comment:
Why you need errors in the post data? Simple create a new variable **$errors = []**